src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -2386,6 +2386,536 @@ void mesh::to_multi_domain(const conduit::Node &n,
     }
 }
 
+//-----------------------------------------------------------------------------
+void
+mesh::connectivity::make_element_2d(std::vector<int64_t>& elem,
+                                    int64_t element,
+                                    int64_t iwidth)
+{
```

Comment:
This always creates one element defined by four vertices.  There is no accumulation nor appending of elements inside this function.

Reply:
`make_element_3d` is different because it has to handle both elements and subelements.  Each call creates a single element, but that element is made up of subelements (faces) that can be shared by two elements.  The `{i|j|k}faces` arguements are there so that the function can first figure out if a needed subelement already exists, then use it if it does, or create a new one and append it if it doesn't.

Reply:
After reviewing your comments and the content of `conduit::blueprint::mesh::connectivity::create_elements_2d`, I understand how this works now. As I have alluded to elsewhere, I feel like this interface suggests that there should be some form of superstructure that encapsulates all of these bits of functionality in a more legible way (e.g. `struct`/`class` a la `ShapeTopology`).